pilot/pkg/config/kube/gateway/testdata/deployment/waypoint.yaml

```diff
@@ -0,0 +1,211 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    gateway.istio.io/managed: istio.io-mesh-controller
+  name: namespace-istio-waypoint
```

Comment:
Can we add a TODO list of bug to track reviewing/changing all user-visible names ? 

I don't want to delay this PR - it is a merge - but the review criteria in an experimental branch is slightly different from the review on main - that's why it's experimental. I expect code has been tested and reviewed for functionality, and it is off by default - but we got into a lot of problems by
reviewing APIs and names as 'experimental/alpha' and then having to move them to beta/stable
due to backwards compat.

Reply:
Yes definitely agree. Opened https://github.com/istio/istio/issues/43437 to track

Reply:
Also to confirm - now or in future we will allow the user to create a ServiceAccount with the same name and additional annotations ? And we won't add or merge this on top ? 

Can you at least add a TODO: that this needs to be documented and discussed. It has to be documented since users may need to add annotations ( for example to allow waypoint to get 3p service account tokens) and used in RBAC rules.

Reply:
The plan is if the resource already exists (without `gateway.istio.io/managed`) we will not touch it, so they can own it themselves. Note this isn't implemented but tracked in https://github.com/istio/istio/issues/43439

Reply:
Note since we use Apply they can add annotations and we won't revert them btw. But it is useful to have them be able to fully own it I think.